sys/net/network_layer/fib/fib_addon_ipv6.c

```diff
@@ -0,0 +1,141 @@
+/**
+ * FIB addon implementation - for ipv6 address types
+ *
+ * Copyright (C) 2015 Martin Landsmann <Martin.Landsmann@HAW-Hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ * @ingroup fib
+ * @{
+ * @file    fib_addon_ipv6.c
+ * @brief   Functions to manage ipv6 FIB entries
+ * @author  Martin Landsmann <Martin.Landsmann@HAW-Hamburg.de>
+ * @}
+ */
+
+#include <errno.h>
+#include "socket_base/socket.h" /**< needed for IN6ADDRSZ and AF_INET6 */
```

Comment:
Since we aim to use our network stack even without sockets I think it would be best to use your own type (or, if that is possible `net/ng_nettype.h`?)